src/python/WMCore/WMSpec/WMWorkload.py

```diff
@@ -324,6 +324,20 @@ def getTask(self, taskName):
             return None
         return WMTaskHelper(task)
 
+    def getTaskByName(self, taskName):
```

Comment:
I've wrote the same method in my PR #7929, though mine seems to never fetch the top level task :) I'm updating it with your code.